### PR TITLE
es6 refactor

### PR DESCRIPTION
**Pull request by JaccoGoris**
I've refactored this module to ES6 and updated the proptypes to use the new `prop-types` module needed by react 16.
I've only tested it with npm link so I think it needs further testing and also new tests written, but I think it's a good start.

### CODE REVIEW

File: lib/WindowResizeListener.js [hunk 1]
```diff
@@ -2,93 +2,115 @@
...
+      document.documentElement.clientWidth ||
+      document.body.clientWidth
+    const windowHeight = window.innerHeight ||
+      document.documentElement.clientHeight ||
+      document.body.clientHeight
+
+    this._listeners.forEach(function (listener) {
+      listener({
+        windowWidth: windowWidth,
+        windowHeight: windowHeight
```
Comment (AndersDJohnson) [inline lib/WindowResizeListener.js]: Could use ES6 shorthand assignment here.

**Review by AndersDJohnson — APPROVED** [review]

**Comment by JaccoGoris** [comment]
Good point, I changed it :D

**Comment by outdooricon** [comment]
@cesarandreu Is there anything else that this pr needs? We really need this merged, the warnings in react are overwhelming right now.

**Comment by kaptron** [comment]
FYI for anyone who finds themselves on this thread, @kunokdev has created an es6 refactored package with updated dependencies, with a corresponding npm package 🙌  
https://github.com/kunokdev/react-window-size-listener

**Pull request closed without merging**